File: semestre-1/disciplina-2-modelagem-de-dados/unidade-3/aula-3/aula-3.java
// Arquivo: Pessoa.java
public class Pessoa {
    private String nome;
    private String endereco;
    private String rg;
    private String cpf;
    private String dataNascimento;
    private String nomeMae;
    private String nomePai;

    public Pessoa(String nome, String endereco, String rg, String cpf, String dataNascimento, String nomeMae, String nomePai) {
        this.nome = nome;
        this.endereco = endereco;
        this.rg = rg;
        this.cpf = cpf;
        this.dataNascimento = dataNascimento;
        this.nomeMae = nomeMae;
        this.nomePai = nomePai;
    }

    // Getters para os atributos
    public String getNome() { return nome; }
    public String getEndereco() { return endereco; }
    // ... e assim por diante para todos os atributos
    
    @Override
    public String toString() {
        return "Nome: " + nome + "\n" +
               "Endereço: " + endereco + "\n" +
               "RG: " + rg + "\n" +
               "CPF: " + cpf + "\n" +
               "Data de Nascimento: " + dataNascimento + "\n" +
               "Nome da Mãe: " + nomeMae + "\n" +
               "Nome do Pai: " + nomePai;
    }
}

// Arquivo: Funcionario.java
public class Funcionario extends Pessoa {
    private String dataAdmissao;
    private String dataDemissao;
    private double salario;

    public Funcionario(String nome, String endereco, String rg, String cpf, String dataNascimento, String nomeMae, String nomePai, String dataAdmissao, String dataDemissao, double salario) {
        super(nome, endereco, rg, cpf, dataNascimento, nomeMae, nomePai); // Chamada ao construtor da superclasse
        this.dataAdmissao = dataAdmissao;
        this.dataDemissao = dataDemissao;
        this.salario = salario;
    }

    @Override
    public String toString() {
        return "--- Dados do Funcionário ---\n" +
               super.toString() + "\n" +
               "Data de Admissão: " + dataAdmissao + "\n" +
               "Data de Demissão: " + dataDemissao + "\n" +
               "Salário: R$" + String.format("%,.2f", salario);
    }
}

// Arquivo: Professor.java
public class Professor extends Pessoa {
    private double valorHoraAula;
    private int horasAula;

    public Professor(String nome, String endereco, String rg, String cpf, String dataNascimento, String nomeMae, String nomePai, double valorHoraAula, int horasAula) {
        super(nome, endereco, rg, cpf, dataNascimento, nomeMae, nomePai);
        this.valorHoraAula = valorHoraAula;
        this.horasAula = horasAula;
    }

    @Override
    public String toString() {
        return "--- Dados do Professor ---\n" +
               super.toString() + "\n" +
               "Valor da Hora/Aula: R$" + String.format("%,.2f", valorHoraAula) + "\n" +
               "Quantidade de Horas/Aula: " + horasAula;
    }
}

// Arquivo: Aluno.java
public class Aluno extends Pessoa {
    private String dataEntrada;
    private String dataFormatura;

    public Aluno(String nome, String endereco, String rg, String cpf, String dataNascimento, String nomeMae, String nomePai, String dataEntrada, String dataFormatura) {
        super(nome, endereco, rg, cpf, dataNascimento, nomeMae, nomePai);
        this.dataEntrada = dataEntrada;
        this.dataFormatura = dataFormatura;
    }
    
    @Override
    public String toString() {
        return "--- Dados do Aluno ---\n" +
               super.toString() + "\n" +
               "Data de Entrada na Faculdade: " + dataEntrada + "\n" +
               "Data de Formatura: " + dataFormatura;
    }
}

// Arquivo: Main.java (para rodar o exemplo)
public class Main {
    public static void main(String[] args) {
        // Criando um objeto da classe Professor
        Professor profExemplo = new Professor(
            "Carlos Martins",
            "Rua das Rosas, 123",
            "12.345.678-9",
            "123.456.789-00",
            "10/05/1980",
            "Maria Silva",
            "José Martins",
            75.00,
            20
        );

        // Criando um objeto da classe Aluno
        Aluno alunoExemplo = new Aluno(
            "Ana Souza",
            "Avenida Brasil, 456",
            "98.765.432-1",
            "987.654.321-00",
            "25/11/2000",
            "Paula Lima",
            "João Souza",
            "03/02/2020",
            "15/12/2024"
        );
        
        // Exibindo os dados
        System.out.println(profExemplo);
        System.out.println("\n" + "=".repeat(30) + "\n");
        System.out.println(alunoExemplo);
    }
}

File: semestre-2/disciplina-2-RENOMEAR/unidade-3/aula-3/aula-3.py
## Arquivo: universidade.py

# A classe Pessoa é a superclasse (generalização)
class Pessoa:
    def __init__(self, nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai):
        self.nome = nome
        self.endereco = endereco
        self.rg = rg
        self.cpf = cpf
        self.data_nascimento = data_nascimento
        self.nome_mae = nome_mae
        self.nome_pai = nome_pai

    def __str__(self):
        return (f"Nome: {self.nome}\nEndereço: {self.endereco}\n"
                f"RG: {self.rg}\nCPF: {self.cpf}\n"
                f"Data de Nascimento: {self.data_nascimento}\n"
                f"Nome da Mãe: {self.nome_mae}\n"
                f"Nome do Pai: {self.nome_pai}")

# A classe Funcionario herda de Pessoa (especialização)
class Funcionario(Pessoa):
    def __init__(self, nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai, data_admissao, data_demissao, salario):
        super().__init__(nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai)
        self.data_admissao = data_admissao
        self.data_demissao = data_demissao
        self.salario = salario
    
    def __str__(self):
        return (f"--- Dados do Funcionário ---\n"
                f"{super().__str__()}\n"
                f"Data de Admissão: {self.data_admissao}\n"
                f"Data de Demissão: {self.data_demissao}\n"
                f"Salário: R${self.salario:,.2f}")

# A classe Professor herda de Pessoa (especialização)
class Professor(Pessoa):
    def __init__(self, nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai, valor_hora_aula, horas_aula):
        super().__init__(nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai)
        self.valor_hora_aula = valor_hora_aula
        self.horas_aula = horas_aula

    def __str__(self):
        return (f"--- Dados do Professor ---\n"
                f"{super().__str__()}\n"
                f"Valor da Hora/Aula: R${self.valor_hora_aula:,.2f}\n"
                f"Quantidade de Horas/Aula: {self.horas_aula}")

# A classe Aluno herda de Pessoa (especialização)
class Aluno(Pessoa):
    def __init__(self, nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai, data_entrada, data_formatura):
        super().__init__(nome, endereco, rg, cpf, data_nascimento, nome_mae, nome_pai)
        self.data_entrada = data_entrada
        self.data_formatura = data_formatura
    
    def __str__(self):
        return (f"--- Dados do Aluno ---\n"
                f"{super().__str__()}\n"
                f"Data de Entrada na Faculdade: {self.data_entrada}\n"
                f"Data de Formatura: {self.data_formatura}")

# --- Criando e exibindo objetos para exemplificar ---

# Criando um objeto da classe Professor
prof_exemplo = Professor(
    nome="Carlos Martins",
    endereco="Rua das Rosas, 123",
    rg="12.345.678-9",
    cpf="123.456.789-00",
    data_nascimento="10/05/1980",
    nome_mae="Maria Silva",
    nome_pai="José Martins",
    valor_hora_aula=75.00,
    horas_aula=20
)

# Criando um objeto da classe Aluno
aluno_exemplo = Aluno(
    nome="Ana Souza",
    endereco="Avenida Brasil, 456",
    rg="98.765.432-1",
    cpf="987.654.321-00",
    data_nascimento="25/11/2000",
    nome_mae="Paula Lima",
    nome_pai="João Souza",
    data_entrada="03/02/2020",
    data_formatura="15/12/2024"
)

# Exibindo os dados
print(prof_exemplo)
print("\n" + "="*30 + "\n")
print(aluno_exemplo)